### ISSUE TITLE
How can I unsubscribe  .onMessage() listener.

### ISSUE THREAD

**Issue by lazyhoneyant** [invertase/react-native-firebase#205]
I wrote following code.

```
this.firebaseMessage = firebase.messaging().onMessage((message) => {
 console.log(message);
});
```

and then To unsubscribe the event

```
  componentWillUnmount(){
    if(this.firebaseMessage) {
      this.firebaseMessage.off();
    }
  }
```

Or 

```
  componentWillUnmount(){
    if(this.firebaseMessage) {
      this.firebaseMessage();
    }
  }
```

But Above codes does not work. 
After digging, I found out following code works.

`this.firebaseMessage.remove();`

Is it the right way?
Commonly in this library, To unsubscribe listener, use off() or call the function returned by event function as I understand.

**Comment by Ehesp** [comment]
Which version are you on? From memory I think this was fixed in 2.0.0: https://github.com/invertase/react-native-firebase/blob/6f9cfd6ba1af680a143246cd7c3183f2fb2228ad/lib/modules/messaging/index.js#L221

**Comment by lazyhoneyant** [comment]
@Ehesp Thanks to the reply. I used 1.1.0. I'm going to migrate.
